web/src/components/base/InputSection.tsx

```diff
@@ -22,47 +20,143 @@ export const Title = styled.label`
   font-weight: bold;
 `
 
+const GeneralInputStyles = css<{ submitted: boolean }>`
+  padding: 0.75rem;
+  background-clip: padding-box;
+  border: 1px solid ${props => props.theme.colors.textDisabledColor};
+  ${props =>
+    props.submitted &&
+    css`
+      :invalid {
+        :focus {
+          outline-color: ${props => props.theme.colors.invalidInput};
+        }
+
+        border-width: 2px;
+        border-color: ${props => props.theme.colors.invalidInput};
+      }
+    `}
+`
+
+export const CompactTitle = styled.label`
+  position: relative;
+  padding: 0 4px;
+  top: 0.9rem;
+  left: 10px;
+  font-size: ${props => props.theme.fonts.decorativeFontSizeSmall};
+  background-color: ${props => props.theme.colors.backgroundColor};
+  width: fit-content;
+`
+
 const StyledTextArea = styled.textarea`
-  resize: none;
+  ${GeneralInputStyles};
+  border-radius: 0.2rem 0.2rem 0;
+  resize: vertical;
   min-height: 60px;
 `
 
-type InputSectionProps = {
+const TextInput = styled.input`
+  ${GeneralInputStyles};
+  border-radius: 0.2rem;
+  background-clip: padding-box;
+  resize: none;
+`
+
+type InputProps = {
   id: string
-  title: string
-  description?: string
   value: string
+  submitted?: boolean
   onChange: (input: string) => void
-  type?: HTMLInputTypeAttribute
   multiline?: boolean
   numberOfLines?: number
-  showOptional?: boolean
+  maxLength?: number
+  required?: boolean
 }
 
-const InputSection = ({
+type InputSectionProps = (
+  | {
+      title: string
+      compact?: false
+      subtitle?: string
+      description?: string
+    }
+  | { title: string; compact: true }
+) &
+  ({ optional: true; showOptional?: boolean } | { optional?: false; showOptional?: false }) &
+  Omit<InputProps, 'required'>
```

Comment:
Why is this complicated type needed? What cases does this prevent?
For example I don't get that title needs to be of type string in both cases and compact can either be true or false or undefined?
Also, why do we need optional, showOptional and required?

Reply:
There are cases where we apparently do not want to show that a text field is optional, even though it is optional. See [this discussion](https://www.figma.com/file/EiRpMhSqscuaChGdhNVZQ4?node-id=34:340&mode=design#593523521) in figma.

Reply:
This type encompasses the following options:
- compact format with only one title (which is also the label)
- non-compact format with title (which is the label) and an optional description and an optional subtitle

And the optional options:
- show that an input field is optional, when it is optional
- do not show that an input is optional, when it is optional
- a text field is required, therefore not optional, show Optional should be undefined or false